treeherder/model/management/commands/cycle_data.py

```diff
@@ -132,15 +136,9 @@ def cycle(self):
     def _remove_leftovers(self):
         # remove any signatures which are
         # no longer associated with a job
-        logger.warning('Removing performance signatures with missing jobs...')
-        for signature in PerformanceSignature.objects.filter(last_updated__lte=self.max_timestamp):
-            self._quit_on_timeout()
-
-            if not PerformanceDatum.objects.filter(
-                repository_id=signature.repository_id,  # leverages (repository, signature) compound index
-                signature_id=signature.id,
-            ).exists():
-                signature.delete()
+        tc_model = TaskclusterModel(root_url, client_id=CLIENT_ID, access_token=ACCESS_TOKEN)
+        signatures_remover = PublicSignaturesRemover(timer=self.timer, taskcluster_model=tc_model)
+        signatures_remover.remove_in_chunks(self.max_timestamp)
```

Comment:
I'd rather prefer we provide the signatures to remove as a queryset instead of the `self.max_timestamp`.
That way, the `PublicSignatureRemover` doesn't have to know which signatures to remove. It will only be concerned with properly removing what the client instructs it to & emailing about it.